operator/internal/status/status.go

```diff
@@ -30,18 +30,26 @@ func Refresh(ctx context.Context, k k8s.Client, req ctrl.Request) error {
 
 	cs := s.Status.Components
 
+	if degraded != nil {
```

Comment:
Would put it before the `cs := ` assignment, so that it is clear the assignment belongs to the instructions below.